wrangler-core/src/main/java/io/cdap/functions/DateAndTime.java

```diff
@@ -0,0 +1,1096 @@
+      .appendLiteral("-")
+      .appendValue(ChronoField.DAY_OF_MONTH, 2)
+      .optionalStart()
+      .appendLiteral(" ")
+      .appendValue(ChronoField.HOUR_OF_DAY, 2)
+      .appendLiteral(':')
+      .appendValue(ChronoField.MINUTE_OF_HOUR, 2)
+      .appendLiteral(':')
+      .appendValue(ChronoField.SECOND_OF_MINUTE, 2)
+      .appendOptional(
+        new DateTimeFormatterBuilder()
+          .appendLiteral('.')
+          .appendFraction(ChronoField.MICRO_OF_SECOND, 0, 6, false)
+          .toFormatter()
+      )
+      .toFormatter();
+
+  private static final DateTimeFormatter TIME_FORMAT =
+    new DateTimeFormatterBuilder()
+      .appendValue(ChronoField.HOUR_OF_DAY, 2)
+      .appendLiteral(':')
+      .appendValue(ChronoField.MINUTE_OF_HOUR, 2)
+      .appendLiteral(':')
+      .appendValue(ChronoField.SECOND_OF_MINUTE, 2)
+      .appendOptional(
+        new DateTimeFormatterBuilder()
+          .appendLiteral('.')
+          .appendFraction(ChronoField.MICRO_OF_SECOND, 0, 6, false)
+          .toFormatter()
+      ).toFormatter();
+
+  private static final DateTimeFormatter OUTPUT_DATETIME_FORMAT
+    = DateTimeFormatter.ofPattern("HH:mm:ss dd MMM yyyy");
+
+  @VisibleForTesting
+  static LocalDate getDate(String date) {
+    return LocalDate.parse(date, DATE_TIME_FORMAT);
+  }
+
+  @VisibleForTesting
+  static LocalTime getTime(String time) {
+    return LocalTime.parse(time, TIME_FORMAT);
+  }
+
+  @VisibleForTesting
+  static LocalDateTime getTimestamp(String date) {
+    return LocalDateTime.parse(date, DATE_TIME_FORMAT);
+  }
+
+  private static DayOfWeek getDayOfWeek(String textDay) {
+    DayOfWeek dayOfWeek = DayOfWeek.SUNDAY;
+    switch(textDay.toUpperCase()) {
+      case "MON":
+      case "MONDAY":
+        dayOfWeek = DayOfWeek.MONDAY;
+        break;
+
+      case "TUE":
+      case "TUESDAY":
+        dayOfWeek = DayOfWeek.TUESDAY;
+        break;
+
+      case "WED":
+      case "WEDNESDAY":
+        dayOfWeek = DayOfWeek.WEDNESDAY;
+        break;
+
+      case "THU":
+      case "THURSDAY":
+        dayOfWeek = DayOfWeek.THURSDAY;
+        break;
+
+      case "FRI":
+      case "FRIDAY":
+        dayOfWeek = DayOfWeek.FRIDAY;
+        break;
+
+      case "SAT":
+      case "SATUDAY":
+        dayOfWeek = DayOfWeek.SATURDAY;
+        break;
+
+      case "SUN":
+      case "SUNDAY":
+        dayOfWeek = DayOfWeek.SUNDAY;
+        break;
+    }
+    return dayOfWeek;
+  }
+
+  /**
+   * @return returns current date with time.
+   */
+  public static LocalDate CurrentDate() {
+    return LocalDate.now();
+  }
+
+  /**
+   * @return return current localtime.
+   */
+  public static LocalTime CurrentTime() {
+    return LocalTime.now();
+  }
+
+  /**
+   * @return return current time in milliseconds.
+   */
+  public static long CurrentTimeMS() {
+    return CurrentTime().getNano() / 1000000L;
+  }
+
+  /**
+   * @return return current data & time.
+   */
+  public static LocalDateTime CurrentTimestamp() {
```

Comment:
should this be a ZonedDateTime? A timestamp usually refers to an exact point in time

Reply:
still have this question.

Reply:
The DataStage date and time functions basically do not operate on a ZonedDateTime, here is their doc: https://www.ibm.com/support/knowledgecenter/SSZJPZ_11.7.0/com.ibm.swg.im.iis.ds.parjob.dev.doc/topics/r_deeref_Date_and_Time_Functions.html. 
We have also heard about use cases about these functions used in LocalDateTime. I feel these functions are more like the Datetime type in Bigquery.

Reply:
This is poor naming though, seeing how we have a 'timestamp' type that is an actual timestamp and not a date time. I think it's better to have a `ZonedDateTime CurrentTimestamp()` and a `LocalDateTime CurrentDateTime()` instead.